ngraph/frontend/onnx_import/src/editor/editor.cpp

```diff
@@ -205,6 +210,35 @@ namespace
             tensor_type->set_elem_type(initializer.data_type());
         }
     }
+
+    // Shift indexes in vectors due to node removal
+    // example:
+    // {2, 4}       {2, 4}
+    // {3, 5, 6} => {3, 4, 5} 4th node will be removed so 5th and 6th node will have a new index
+    // {0, 1}       {0, 1}
+    void shift_node_indexes(std::vector<std::vector<int>>& node_indexes)
+    {
+        for (int i = node_indexes.size() - 1; i >= 1; --i)
+        {
+            std::vector<int> shifts(node_indexes[i].size());
+            for (int j = 0; j < i; ++j)
+            {
+                int dim = 0;
+                for (auto index : node_indexes[i])
+                {
+                    auto shift = std::count_if(node_indexes[j].begin() + 1,
+                                               node_indexes[j].end(),
+                                               [&index](int val) { return val < index; });
+                    shifts[dim++] += shift;
```

Comment:
Should it be `+=` or just `=` instead? I don't think the same element is modified more than once and thus the addition is not needed here.

Reply:
it should be `+=` because multiple vectors can affect this shift.
example:
{0, 1} << it will be not affected because it is a first vector
{2, 3, 4} << node no. 1 will be removed so shifts will have {1, 1, 1} value
{5, 6, 7} << node no. 1 will be removed so shifts = {1, 1, 1) but nodes 3 and 4 will also be removed so shifts will be increased to {3, 3, 3}
final values:
{0, 1}
{1, 2, 3}
{2, 3, 4}